Sources/PromotedCore/MetricsLogger.swift

```diff
@@ -219,7 +231,7 @@ fileprivate extension MetricsLogger {
   private func userInfoMessage() -> Common_UserInfo {
     var userInfo = Common_UserInfo()
     if let id = userID { userInfo.userID = id }
-    userInfo.logUserID = logUserID
+    if let id = internalLogUserID { userInfo.logUserID = id }
```

Comment:
does the if conditional assignment only fail on null assignment only, or could it treat some values of id as false as well?

Reply:
It fails on null only.